medcat/cat.py

```diff
@@ -604,7 +616,7 @@ def unlink_concept_name(self, cui: str, name: str, preprocessed_name: bool = Fal
 
         cuis = [cui]
         if preprocessed_name:
-            names = {name: 'nothing'}
+            names = {name: {'nothing': 'nothing'}}
```

Comment:
interesting - so this was a bug before?

Reply:
The issue had to do with later calling `CDB.remove_names` with the `names` argument.
The `names` could either have the type `Dict[str, Set[str]]` (if `preprocessed_name`) or `Dict[str, Dict]` otherwise from `medcat.preprocessing.cleaners.prepare_name` (although the type hasn't been explicitly type-hinted by the method return type).

Before I started this procedure, the type hint for the `CDB.remove_names` method's `names` argument was simply `Dict`. So that worked with either of the aforementioned inputs.
However, the doc string said `Dict[str, Dict]`. So since `flake8` was now checking doc strings, I assumed the more restrictive type would be appropriate and changed the type hint for the method.
After having made all the doc string changes, `mypy` was now having issues for the argument type. Which is why I made this change.

With that said, the `CDB.remove_names` method only acts on the keys of this dict. So it doesn't really matter what the values are. It's just that in most cases, the results from `medcat.preprocessing.cleaners.prepare_name` would be expected to be provided.

So all in all not necessarily a bug. Behaviour isn't changed, after all. But rather a clarification of the types of data generally expected.

Reply:
yeah that makes more sense . slightly less confusing